engine/src/objectprops.cpp

```diff
@@ -724,7 +724,7 @@ void MCObject::setvariantprop(MCExecContext& ctxt, uint32_t p_part_id, Propertie
 void MCObject::setdataprop(MCExecContext& ctxt, uint32_t p_part_id, Properties p_which, Boolean p_effective, MCDataRef p_value)
 {
     MCExecValue t_value;
-    t_value . dataref_value = p_value;
```

Comment:
Take this change out of this PR - we should file a separate bug and fix separately otherwise we risk introducing a bug unrelated to the bug this PR is meant to fix.